schema/fields.go

```diff
@@ -477,3 +478,34 @@ func buildQueryableField(parent string, f *Field) *QueryableField {
 
 	return NewQueryableField(name, f.Type())
 }
+
+func BuildPartitionFields(schema jsoniter.RawMessage, fields []*Field) ([]*Field, error) {
+	var partitionFields []*Field
+
+	val, dt, _, err := jsonparser.Get(schema, PartitionKeyF)
+	if dt == jsonparser.NotExist {
+		return partitionFields, nil
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if dt == jsonparser.Array {
+		var keys []string
+		if err := jsoniter.Unmarshal(val, &keys); err != nil {
+			return nil, err
+		}
+
+		for _, key := range keys {
+			for _, field := range fields {
+				if key == field.FieldName {
+					partitionFields = append(partitionFields, field)
```

Comment:
field can have an attribute partition key, similar to the primary key that we can set.